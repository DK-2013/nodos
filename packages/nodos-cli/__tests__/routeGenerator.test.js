import fsp from 'fs/promises';
import path from 'path';
import os from 'os';
import generateNewRoute from '../lib/current/routeGenerator.js';

// let dir;
const routePath = path.join(__dirname, '../__fixtures__/site/config/routes.yml');

let dir;

beforeEach(async () => {
  dir = await fsp.mkdtemp(path.join(os.tmpdir(), 'nodos-'));
  const configPath = path.join(dir, 'config');

  await fsp.mkdir(configPath);
  fsp.copyFile(routePath, path.join(configPath, 'routes.yml'));
});

test('nodos/generate/resources', async () => {
  const result = await generateNewRoute(dir, 'resources', 'users');
  expect(result).toMatchSnapshot();
});

test('nodos/generate/resource', async () => {
  const result = await generateNewRoute(dir, 'resource', 'session');
  expect(result).toMatchSnapshot();
});
