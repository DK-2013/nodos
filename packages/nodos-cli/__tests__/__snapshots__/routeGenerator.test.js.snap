// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nodos/generate/resource 1`] = `
"---

pipelines:
  browser: []

scopes:
  - path: /
    pipeline: browser
    routes:
      - root: home#index
      - resource: session"
`;

exports[`nodos/generate/resources 1`] = `
"---

pipelines:
  browser: []

scopes:
  - path: /
    pipeline: browser
    routes:
      - root: home#index
      - resources: users"
`;
